AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -981,6 +981,28 @@ - (void)setContentsScaleForDisplay:(CGFloat)contentsScaleForDisplay
   _contentsScaleForDisplay = contentsScaleForDisplay;
 }
 
+- (void)applyPendingViewState
+{
+  ASDisplayNodeAssertMainThread();
+  ASDN::MutexLocker l(_propertyLock);
+
+  // FIXME: Ideally we'd call this as soon as the node receives -setNeedsLayout
```

Comment:
Totally agree! I only hesitate to do it in this PR because we're already 2500 lines from shore.